src/orders/components/OrderTransaction/utils.ts

```diff
@@ -16,3 +23,121 @@ export const mapActionToMessage: Record<
   CANCEL: transactionActionMessages.cancel,
   REFUND: null,
 };
+
+export const mapTransactionEvent = (
+  event: TransactionEventFragment | TransactionFakeEvent,
+): TransactionMappingResult => {
+  if (!event) {
+    return {
+      type: null,
+      status: null,
+    };
+  }
+
+  if (event.__typename === "TransactionFakeEvent") {
+    return event.mappedResult;
+  }
+
+  switch (event.type) {
```

Comment:
suggestion: istead of having this long switch, IMO simple map object will solve problem